### PR TITLE
Add the guards for partition table involved subqueries other than SPSP

### PR DESCRIPTION
**Pull request by jxchinaren**

### CODE REVIEW

File: src/frontend/org/voltdb/planner/PlanAssembler.java [hunk 1]
```diff
@@ -379,6 +379,9 @@ else if (plan.hasLimitOrOffset()) {
      * @param parsedStmt Current SQL statement to generate plan for
      * @return The best cost plan or null.
      */
+    public static String IN_EXISTS_SCALAR_ERROR_MESSAGE = "Subquery expressions are only supported for "
+            + "single partition procedures and AdHoc replicated tables";
```
Comment (wolffcm) [inline src/frontend/org/voltdb/planner/PlanAssembler.java]: Maybe change this message to "... and AdHoc queries referencing only replicated tables."

File: src/frontend/org/voltdb/planner/PlanAssembler.java [hunk 2]
```diff
@@ -400,6 +403,27 @@ public CompiledPlan getBestCostPlan(AbstractParsedStmt parsedStmt) {
...
+                        m_recentErrorMsg = IN_EXISTS_SCALAR_ERROR_MESSAGE;
+                        return null;
+                    }
+                }
+
+                // no partition tables in subqueries
+                for (AbstractExpression e: subqueryExprs) {
+                    assert(e instanceof SelectSubqueryExpression);
+                    SelectSubqueryExpression subExpr = (SelectSubqueryExpression)e;
+                    if (! subExpr.getSubqueryScan().getIsReplicated()) {
```
Comment (wolffcm) [inline src/frontend/org/voltdb/planner/PlanAssembler.java]: Do we support nesting subqueries more than 1 level?  You could imagine needing to use recursion here.

**Comment by wolffcm** [comment]
That's the end of my comments.  I'd like to understand if multi-level nesting is possible.  Other than that, looks good to me.

**Comment by jxchinaren** [comment]
@wolffcm We do support nested subqueries more than 1 level. Our planning algorithm is deep first planning, so the code does cover the cases. 

Anyway, I can add another planner tests. Also, I realize we do not have enough coverage on multiple level nested subqueries and paralleled subqueries tests, and update the ticket ENG-8102 yesterday.

**Comment by jxchinaren** [comment]
@wolffcm I have an additional commit resolved your comments.

**Comment by wolffcm** [comment]
Looks good.

**Pull request merged**